src/NetMQ/Global.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NetMQ.Core;
+
+namespace NetMQ
+{
+    public static class Global
+    {
+        private static int s_ioThreadCount = 1;
+        private static int s_maxSockets = 1024;
+
+        private static Ctx s_ctx;
+        private static object s_sync = new object();
+        private static int s_socketCount = 0;
+
+        /// <summary>
+        /// Get or set the number of IO Threads in the context, default is 1.
+        /// 1 is good for most cases.
+        /// </summary>
+        public static int ThreadPoolSize
+        {
+            get
+            {
+                lock (s_ctx)
+                {
+                    return s_ioThreadCount;
+                }                
+            }
+            set
+            {
+                lock (s_ctx)
+                {
+                    s_ioThreadCount = value;
+
+                    if (s_ctx != null)
+                    {
+                        s_ctx.IOThreadCount = s_ioThreadCount;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get or set the maximum number of sockets.
+        /// </summary>
+        public static int MaxSockets
+        {
+            get
+            {
+                lock (s_ctx)
+                {
+                    return s_maxSockets;
+                }
+            }
+            set
+            {
+                lock (s_ctx)
+                {
+                    s_maxSockets = value;
+
+                    if (s_ctx != null)
+                    {
+                        s_ctx.MaxSockets = s_maxSockets;
+                    }
+                }
+            }
+        }
+
+        public static bool IsRunning
+        {
+            get
+            {
+                lock (s_sync)
+                {
+                    return s_ctx != null;
+                }
+            }
+        }
+
+        internal static SocketBase CreateSocket(ZmqSocketType socketType)
+        {
+            lock (s_sync)
+            {
+                if (s_ctx == null)
+                {
+                    s_ctx = new Ctx();
```

Comment:
So essentially this is a singleton, and this "guarantees" there is only one context per process, but it's handled behind the scene. No need for explicit NetMQContext unless more than one per process was needed for some reason. Do I understand correctly? Thanks.